Rubberduck.Core/UnitTesting/TestEngine.cs

```diff
@@ -89,66 +89,110 @@ public void Run(IEnumerable<TestMethod> tests)
 
         private void RunInternal(IEnumerable<TestMethod> tests)
         {
-            var testMethods = tests as IList<TestMethod> ?? tests.ToList();
-            if (!testMethods.Any())
+            if (_state.Status != ParserState.Ready)
             {
                 return;
             }
```

Comment:
This is not really compatible with how the tests are run: they are wired up to run on the state change to `ResolvedDeclarations`.

Reply:
I wasn't aware that we automatically run tests on `ResolvedDeclarations`. I will review and revise accordingly.

Reply:
Having reviewed the code I see that tests are permitted to run on `ResolvedDeclaration` and `ResolvingReferences` in addition to the `Ready`. 

Is there any reason to allow suspending of parser in those other states? 

Curiously, when I ran the project, the tests were not run on a refresh, which explains why I did not observe any errors when I made the changes. That needs to be investigated.